### ISSUE TITLE
Year is not recognized correctly 

### ISSUE THREAD

**Issue by vanabel** [jhpalmieri/bibweb#3]
The citation `\nocite{Wolfson;Harmonic;Differential-Geometry;1988}` return two results
```bib
@comment   
@comment   citation 'Wolfson;Harmonic;Differential-Geometry;1988' 
@comment   Math Reviews number: 918462 
@article {MR918462,
    AUTHOR = {Wolfson, Jon G.},
     TITLE = {Harmonic sequences and harmonic maps of surfaces into complex
              {G}rassmann manifolds},
   JOURNAL = {J. Differential Geom.},
  FJOURNAL = {Journal of Differential Geometry},
    VOLUME = {27},
      YEAR = {1988},
    NUMBER = {1},
     PAGES = {161--178},
      ISSN = {0022-040X},
   MRCLASS = {58E20},
  MRNUMBER = {918462},
MRREVIEWER = {Barbara Tabak},
       URL = {http://projecteuclid.org/euclid.jdg/1214441656},
}
@comment   Math Reviews number: 862045 
@article {MR862045,
    AUTHOR = {Wolfson, Jon G.},
     TITLE = {Harmonic maps of the two-sphere into the complex hyperquadric},
   JOURNAL = {J. Differential Geom.},
  FJOURNAL = {Journal of Differential Geometry},
    VOLUME = {24},
      YEAR = {1986},
    NUMBER = {2},
     PAGES = {141--152},
      ISSN = {0022-040X},
   MRCLASS = {58E20},
  MRNUMBER = {862045},
MRREVIEWER = {G. T\'{o}th},
       URL = {http://projecteuclid.org/euclid.jdg/1214440432},
}
```
Clearly, we expect the first one, since the second one is published at **1986**. So It seems like a bug?

I would really appreciate it, if you can find out the reason and fix it.

**Comment by jhpalmieri** [comment]
The reason for this is that currently, the 'year' argument is completely ignored by the code. I don't know if this is just a bug or if there was a reason — there are comments that using years can slow things down quite a bit, so maybe that led me to intentionally disable it a long time ago. I've spent a few minutes tinkering with the code, and I cannot figure out how to get MathSciNet to process the year correctly.

Part of the issue is that, unfortunately, the required interface  to MathSciNet is not documented anywhere. I do not know how to fix this particular problem.

**Comment by vanabel** [comment]
I agree that MathSciNet is not documented anywhere. The only way to investigate is use the browser's development tool to see the request header/url. For example, I find the url:
`https://www.ams.org/mathscinet/2006/mathscinet/search/publications.html?pg4=AUCN&s4=Wolfson&co4=AND&pg5=TI&s5=Harmonic&co5=AND&pg6=JOUR&s6=Differential+and+Geometry&co6=AND&pg7=ALLF&s7=&co7=AND&dr=pubyear&yrop=eq&arg3=1988&yearRangeFirst=&yearRangeSecond=&pg8=ET&s8=All&review_format=html&Submit=Search&fmt=bibtex`
Thus, I fix the year by modify:
```perl
    # parse year entry
    YEAR: {
      $dr = '&yearRangeFirst=&yearRangeSecond=&pg8=ET&s8=All&review_format=html';
      # last YEAR;
      if ($year =~ /^(\d{4})-(\d{4})$/) {
	$yr = '&dr=pubyear&yrop=&arg3=';
	$dr = '&yearRangeFirst='. $1 .'&yearRangeSecond=' . $2 . '&pg8=ET&s8=All&review_format=html';
      }else{
	$relation_year=substr($year, 0, 1); 
	if ($relation_year eq '<' ) {
	  $yr = '&dr=pubyear&yrop=lt&arg3=' . substr($year, 1);
	}
	elsif ($relation_year eq '>' ) {
	  $yr = '&dr=pubyear&yrop=gt&arg3=' . substr($year, 1);
	}
	else {
	  $yr = '&dr=pubyear&yrop=eq&arg3=' . $year;
	}
      }
      $yr = $yr . $dr;
    }

    # construct URL
    $url = "https://$e_math/mathscinet/2006/mathscinet/search/publications.html?" .
	"pg4=AUCN&s4=" .
	"$author" .
	"&co4=AND&pg5=TI&s5=" .
	"$title" .
	"&co5=AND&pg6=JOUR&s6=" .
	"$journal" .
	"&co6=AND&pg7=ALLF&s7=" .
	"$misc" .
	"&co7=AND" .
	"$yr" . "&Submit=Search" . 
	"&fmt=bibtex"	;
```
This will support year like:`>1988`, `<1988`, `=1988` (same as `1988`) and `1987-1988`(the year from 1987 to 1988).

**Comment by vanabel** [comment]
> The reason for this is that currently, the 'year' argument is completely ignored by the code. I don't know if this is just a bug or if there was a reason — there are comments that using years can slow things down quite a bit, so maybe that led me to intentionally disable it a long time ago. I've spent a few minutes tinkering with the code, and I cannot figure out how to get MathSciNet to process the year correctly.
> 
> Part of the issue is that, unfortunately, the required interface to MathSciNet is not documented anywhere. I do not know how to fix this particular problem.



> The reason for this is that currently, the 'year' argument is completely ignored by the code. I don't know if this is just a bug or if there was a reason — there are comments that using years can slow things down quite a bit, so maybe that led me to intentionally disable it a long time ago. I've spent a few minutes tinkering with the code, and I cannot figure out how to get MathSciNet to process the year correctly.
> 
> Part of the issue is that, unfortunately, the required interface to MathSciNet is not documented anywhere. I do not know how to fix this particular problem.

I agree that MathSciNet is not documented anywhere. The only way to investigate is use the browser's development tool to see the request header/url. For example, I find the url:
`https://www.ams.org/mathscinet/2006/mathscinet/search/publications.html?pg4=AUCN&s4=Wolfson&co4=AND&pg5=TI&s5=Harmonic&co5=AND&pg6=JOUR&s6=Differential+and+Geometry&co6=AND&pg7=ALLF&s7=&co7=AND&dr=pubyear&yrop=eq&arg3=1988&yearRangeFirst=&yearRangeSecond=&pg8=ET&s8=All&review_format=html&Submit=Search&fmt=bibtex`
Thus, I fix the year by modify:
```perl
    # parse year entry
    YEAR: {
      $dr = '&yearRangeFirst=&yearRangeSecond=&pg8=ET&s8=All&review_format=html';
      # last YEAR;
      if ($year =~ /^(\d{4})-(\d{4})$/) {
	$yr = '&dr=pubyear&yrop=&arg3=';
	$dr = '&yearRangeFirst='. $1 .'&yearRangeSecond=' . $2 . '&pg8=ET&s8=All&review_format=html';
      }else{
	$relation_year=substr($year, 0, 1); 
	if ($relation_year eq '<' ) {
	  $yr = '&dr=pubyear&yrop=lt&arg3=' . substr($year, 1);
	}
	elsif ($relation_year eq '>' ) {
	  $yr = '&dr=pubyear&yrop=gt&arg3=' . substr($year, 1);
	}
	else {
	  $yr = '&dr=pubyear&yrop=eq&arg3=' . $year;
	}
      }
      $yr = $yr . $dr;
    }

    # construct URL
    $url = "https://$e_math/mathscinet/2006/mathscinet/search/publications.html?" .
	"pg4=AUCN&s4=" .
	"$author" .
	"&co4=AND&pg5=TI&s5=" .
	"$title" .
	"&co5=AND&pg6=JOUR&s6=" .
	"$journal" .
	"&co6=AND&pg7=ALLF&s7=" .
	"$misc" .
	"&co7=AND" .
	"$yr" . "&Submit=Search" . 
	"&fmt=bibtex"	;
```
This will support year like:`>1988`, `<1988`, `=1988` (same as `1988`) and `1987-1988`(the year from 1987 to 1988).

**Comment by jhpalmieri** [comment]
That looks very good, thank you. Fixed by PR #4.

**Comment by vanabel** [comment]
> That looks very good, thank you. Fixed by PR #4.

Thank you for highlighting my fix in your PR comment! I really appreciate your recognition, and I'm glad I could contribute to improving the tool. 

In fact, I'm using it a lot in my recent scientific writing. Let's continue collaborating to make this project even better!
